File: 1041.质数判断/main.cpp
#include <iostream>
#include "cmath"

using namespace std;

int main()
{
    int num,i;
    cin>>num;
    if(num==0||num==1)
    {
        cout<< "No" <<endl;
        return 0;
    }
    for(i=2;i<=sqrt(num);i++)
    {
        if(num%i==0)
        {
            cout<< "No" <<endl;
            return 0;
        }
    }
    cout<< "Yes" <<endl;
    return 0;
}

File: 1041.质数判断/Main.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        System.out.println(
                isPrime(new Scanner(System.in).nextInt())?
                        "Yes":
                        "No"
        );
    }

    public static boolean isPrime(int num){
        if(num<=1)
            return false;
        for(int i=2;i<=((num+1)/2);i++){
            if(num%i == 0)
                return false;
        }
        return true;
    }
}
